### PR TITLE
html-css-popup review Hooli-style Popup

### PR DESCRIPTION
**Pull request by AndriiZahumnyi**
### html-css-popup

[Demo] https://andriizahumnyi.github.io/HTML-CSS-practice-Hooli-style-Popup/ 
[Code base] https://github.com/AndriiZahumnyi/HTML-CSS-practice-Hooli-style-Popup

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by smillims** [comment]
Hey, good work! But I find some problems, let's correct it together!

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 1]
```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
+<html>
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:2]: Add to `html` attribute `lang` to declare the language of this document.

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            <input id="check01" type="checkbox" name="menu" />
+            <label for="check01"><img class="popup_png" src="/popup-icons/popup-button.png" alt="popup button"
+                tabindex="1"></label>
+            <div class="sub_menu">
+              <div class="sub_container">
+                <a href="#">
+                  <div class="align_self" tabindex="1">
+                    <img src="/popup-icons/calendar.png" alt="calendar icon">
+                    <p>Calendar</p>
+                  </div>
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:28]: You can't add block tag `div` inside inline-element `a`.
If you want turn on focus on your page you should style this in your css file with help pseudo-class `:focus`.
I think that the last word in attribute `alt` isn't needed in this case. Just "Calendar" and in the feature cases the same example.

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                </a>
+                <a href="#">
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:30]: Wrap your block .sub_container in `ul` | `ol` instead `div`, after this wrap your links in `li`.
We're practicing here for [ally](https://medium.com/web-standards/pragmatic-a11y-rules-b16dd2f77685), so don't forget about semantic.

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+              <label for="check02" class="more_behavior">More</label>
+            </div>
+          </li>
+          <li><img class="bell_png" src="/popup-icons/bell.png" alt="favicon bell"></li>
+          <li><img class="kottans_png" src="/popup-icons/favicon.png" alt="favicon kottans"></li>
+        </ul>
+      </div>
+    </nav>
+  </header>
+  <!-- <script src="" async defer></script> -->
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:149]: Delete, please

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+</body>
+
+</html>
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:152]: Here's add empty line

File: submissions/AndriiZahumnyi/html-css-popup/main_style.css [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+/*show the more_sub_menu when input is checked*/
+#check02:checked~.more_sub_menu {
+  display: block;
+}
+
+#check02:checked~.more_behavior {
+  display: none;
+}
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/main_style.css:37]: Please, avoid styling by id.

File: submissions/AndriiZahumnyi/html-css-popup/main_style.css [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+.more_sub_menu p {
+  text-align: center;
+}
+
+@media screen and (min-width: 700px) {
+  ul.main_menu li {
+    padding: 1em;
+  }
+}
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/main_style.css:189]: Don't forget about empty line

File: submissions/AndriiZahumnyi/html-css-popup/main_style.css [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+.more_sub_menu p {
+  text-align: center;
+}
+
+@media screen and (min-width: 700px) {
+  ul.main_menu li {
+    padding: 1em;
+  }
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/main_style.css:188]: I recommend (when your `media` some bigger) add to another file and connect with help `link`

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+                <a href="#">
+                  <div class="align_self" tabindex="1">
+                    <img src="/popup-icons/documents.png" alt="documents icon">
+                    <p>Documents</p>
+                  </div>
+                </a>
+              </div>
+              <input id="check02" type="checkbox" name="more_menu" />
+              <div class="more_sub_menu">
+                <hr />
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:99]: I recommend use pseudo-element `:before` or `:after` if you want to add line, but in this case you are free in your choise

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+                  </div>
+                </a>
+                <a href="#">
+                  <div class="align_self" tabindex="1">
+                    <img src="/popup-icons/documents.png" alt="documents icon">
+                    <p>Documents</p>
+                  </div>
+                </a>
+              </div>
+              <input id="check02" type="checkbox" name="more_menu" />
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:97]: Use class, not id

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="main_style.css">
+</head>
+<body>
+  <header class="header">
+    <nav class="head_nav">
+      <div class="container">
+        <ul class="main_menu">
+          <li>Mail</li>
+          <li>Images</li>
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:17]: "Mail" and "Images" must be links

File: submissions/AndriiZahumnyi/html-css-popup/index.html [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+                    </div>
+                  </a>
+                </div>
+                <p><a href="https://kottans.org/" target="_blank">Even more from Kottans</a></p>
+              </div>
+              <label for="check02" class="more_behavior">More</label>
+            </div>
+          </li>
+          <li><img class="bell_png" src="/popup-icons/bell.png" alt="favicon bell"></li>
+          <li><img class="kottans_png" src="/popup-icons/favicon.png" alt="favicon kottans"></li>
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/index.html:144]: I think it's too should be links. Don't forget about `:focus`

File: submissions/AndriiZahumnyi/html-css-popup/main_style.css [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+ul.main_menu {
+  margin: 5px 0;
+}
+
+ul.main_menu li {
+  list-style-type: none; 
+  padding: 0.5em;
+}
```
Comment (smillims) [inline submissions/AndriiZahumnyi/html-css-popup/main_style.css:66]: Add `cursor: poiner`

**Review by smillims — CHANGES_REQUESTED** [review]
Against, here we are practice for [ally](https://medium.com/web-standards/pragmatic-a11y-rules-b16dd2f77685). Your popup-menu and button "More" don't work for using there just with keyboard help, if someone will be need to move in your web-site with keybord help (tab and space), so they couldn't do it.

**Comment by AndriiZahumnyi** [comment]
Hello!
Thanks a lot for review, for your time!
I`ll correct all these things soon and report.

**Review by miroslav-kolomiets — CHANGES_REQUESTED** [review]
Good job!
But I recommend you check your html code here https://validator.w3.org/nu/#textarea
you have many validation errors, please check it and fix it.

**Comment by AndriiZahumnyi** [comment]
Hello, mentors!
I think, I`ve fixed all the problems, that I had in my practice project. Could you review again, please!

**Comment by smillims** [comment]
> Hello, mentors!
> I think, I`ve fixed all the problems, that I had in my practice project. Could you review again, please!

Hey! I think you forget update your code base

**Comment by AndriiZahumnyi** [comment]
> > Hello, mentors!
> > I think, I`ve fixed all the problems, that I had in my practice project. Could you review again, please!
> 
> Hey! I think you forget update your code base

100%!) sorry

**Comment by AndriiZahumnyi** [comment]
> > Hello, mentors!
> > I think, I`ve fixed all the problems, that I had in my practice project. Could you review again, please!
> 
> Hey! I think you forget update your code base

Code base already updated!

**Comment by smillims** [comment]
I like your new correct! Now wait for @miroslav-kolomiets

**Comment by AndriiZahumnyi** [comment]
> I like your new correct! Now wait for @miroslav-kolomiets

OK! Thanks a lot!)

**Review by miroslav-kolomiets — APPROVED** [review]
Good job! Keep going!

**Comment by AndriiZahumnyi** [comment]
> Good job! Keep going!

Thanks a lot!!

**Pull request merged**